src/main/java/space/obminyashka/items_exchange/util/QPredicate.java

```diff
@@ -0,0 +1,32 @@
+package space.obminyashka.items_exchange.util;
+
+import com.querydsl.core.types.ExpressionUtils;
+import com.querydsl.core.types.Predicate;
+import com.querydsl.core.types.dsl.Expressions;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.util.*;
+import java.util.function.Function;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class QPredicate {
+
+    private final Set<Predicate> predicates = new HashSet<>();
+
+    public <T> QPredicate add(T object, Function<T, Predicate> function) {
+        if (object instanceof Collection && !((Collection<?>) object).isEmpty()) {
+            predicates.add(function.apply(object));
+        }
+        return this;
+    }
+
+    public Predicate buildAnd() {
+        Predicate predicate = ExpressionUtils.allOf(predicates);
+        return predicate == null ? Expressions.asBoolean(true).isTrue() : predicate;
```

Comment:
You can use **BooleanUtils.toBooleanDefaultIfNull()** and inline that expression

Reply:
Doesn't suit :(